app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsFragment.kt

```diff
@@ -434,6 +438,27 @@ class NotificationsFragment :
         }
     }
 
+    private fun removeDuplicateOlderEntries(positionStart: Int) {
+        val dataList = adapter.snapshot().items
+
+        Log.w(TAG, "found elements in adapter "+dataList.size)
+
+        for (pos in dataList.size - 1 downTo  positionStart) {
+            val notificationViewData = dataList[pos]
+
+            val status = notificationViewData.statusViewData?.status
+                ?: continue
+
+            if (!viewModel.hasNewestNotificationId(notificationViewData.type, status.id, notificationViewData.id)) {
+                Log.w(TAG, "Removing old notification at "+pos+" for "+status.id)
+
+                adapter.notifyItemRemoved(pos)
```

Comment:
Simply "notify removed" seems odd as I don't have removed something but I want it to be removed in the list.

However I guess it might be correct as the backing list is api calls and what is presented in the list is or should not be exactly the api data.

Reply:
This is a good argument for doing this in the viewmodel, so that the data removal happens before the data is inserted in to the adapter (per my previous comment).

Reply:
There are two types of filtering here:

One happens in the view model (`pagingData.filter`) but the one here is to remove old entries that are made only obsolete by other new entries.